aws/resource_aws_wafv2_ip_set.go

```diff
@@ -0,0 +1,277 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/wafv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsWafv2IPSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafv2IPSetCreate,
+		Read:   resourceAwsWafv2IPSetRead,
+		Update: resourceAwsWafv2IPSetUpdate,
+		Delete: resourceAwsWafv2IPSetDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				idParts := strings.Split(d.Id(), "/")
+				if len(idParts) != 3 || idParts[0] == "" || idParts[1] == "" || idParts[2] == "" {
+					return nil, fmt.Errorf("Unexpected format of ID (%q), expected ID/NAME/SCOPE", d.Id())
+				}
+				id := idParts[0]
+				name := idParts[1]
+				scope := idParts[2]
+				d.SetId(id)
+				d.Set("name", name)
+				d.Set("scope", scope)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"addresses": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MinItems: 1,
```

Comment:
```suggestion
```
hi @pvanbuijtene 👋, we'll want to allow for an empty list here else if a user provides in a config `addresses = []`, there will an error at plan time to enforce the 1 item minimum